lib/parsers/lock-parser-base.ts

```diff
@@ -382,16 +395,33 @@ export abstract class LockParserBase implements LockfileParser {
       const depKey = depOrder.shift() as string;
       const dep = depMap[depKey];
       let treeSize = 1;
-
+      if (!dep) {
+        debug(`Missing entry for ${depKey}`);
+        continue;
+      }
       // direction is from the dependency to the package requiring it, so we are
       // looking for predecessors
       for (const subDepPath of depGraph.predecessors(depKey)) {
-        const subDep = depTrees[subDepPath];
+        let subDep = depTrees[subDepPath];
         if (!dep.dependencies) {
           dep.dependencies = {};
         }
+        if (!subDep) {
+          debug(`Missing entry for ${subDepPath}`);
+          const { name, identifier } = extractNameAndIdentifier(subDepPath);
+          subDep = {
```

Comment:
create an empty dep with out of sync label to follow the standard we have for missing top level deps.